File: frontend/src/app/interfaces/valores-evento.dto.ts
/**
 * Valores por defecto de los eventos de la peña. Cualquiera puede venir a null: significa que ese
 * campo no se sugiere y el formulario del evento lo deja vacío.
 */
export interface ValoresEvento {
  /** Plazas del evento (el autobús, normalmente). */
  plazas?: number | null;
  /** Lo que paga cada socio por su plaza. */
  costePlaza?: number | null;
  /** Carnets que se sortean. */
  carnets?: number | null;
  /** Lo que paga quien se lleva un carnet. */
  costeCarnet?: number | null;
  /** Coste total estimado del evento, para las cuentas de la peña. */
  costeTotalEstimado?: number | null;
  /**
   * Días antes del evento en que cierra la inscripción, y la hora de ese día ('HH:mm'). Se
   * guardan relativos porque lo que se repite de un partido a otro no es una fecha concreta sino
   * "dos días antes, a las ocho".
   */
  diasAntesFinInscripcion?: number | null;
  horaFinInscripcion?: string | null;
  /** Lo mismo para el sorteo de carnets. */
  diasAntesSorteo?: number | null;
  horaSorteo?: string | null;
}

File: src/main/java/com/softwells/fanops/controller/dto/ValoresEventoDTO.java
package com.softwells.fanops.controller.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import java.math.BigDecimal;
import java.time.LocalTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Valores por defecto de los eventos de una peña. Todos los campos pueden venir a null: significa
 * que ese campo no se sugiere y el formulario lo deja vacío.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ValoresEventoDTO {

  private Integer plazas;
  private BigDecimal costePlaza;
  private Integer carnets;
  private BigDecimal costeCarnet;
  private BigDecimal costeTotalEstimado;

  /** Días antes del evento en que cierra la inscripción, y a qué hora. */
  private Integer diasAntesFinInscripcion;

  @JsonFormat(pattern = "HH:mm")
  private LocalTime horaFinInscripcion;

  /** Días antes del evento en que se celebra el sorteo, y a qué hora. */
  private Integer diasAntesSorteo;

  @JsonFormat(pattern = "HH:mm")
  private LocalTime horaSorteo;
}
